src/components/container/BloodPressureVisualization/BloodPressureVisualization.tsx

```diff
@@ -0,0 +1,303 @@
+                var db : Dumbbell = {dataPoint  :dataPoint, xValue : logDay, class : assignClass(systolicAverage, diastolicAverage)};
+                dbs.push({date: entriesForDate[0].date, dumbbell: db, averageSystolic : systolicAverage, averageDiastolic : diastolicAverage});
+            });
+        }
+
+        return dbs;
+    }
+
+    function pageWeeklyData(startOfWeek: Date, freshData? : BloodPressureDumbbell[]){
+        const useData = freshData ?? bpData;
+        const weekData : Dumbbell[] = [];
+        const weekDataForMetrics : BloodPressureDumbbell[] = [];
+        for (let i = 0; i < 7; i++) {
+            var currentDate = startOfDay(addDays(startOfWeek, i));
+            var dataForDay = useData.find( db => isEqual(db.date, currentDate));
+            if (!dataForDay)
+            {
+                var formattedDate = format(currentDate, "MM/dd");
+                weekData.push({xValue : formattedDate});
+            } 
+            else 
+            {
+                weekDataForMetrics.push(dataForDay);
+                weekData.push(dataForDay.dumbbell);
+            }
+        }
+
+        const weeklyMetrics = getWeeklyAggregates(weekDataForMetrics);
+        setMetrics(weeklyMetrics);
+        setWeeklyDataForViz(weekData);
+        setStartOfWeek(startOfWeek);
+    }
+
+    function getWeeklyAggregates(dbs : BloodPressureDumbbell[]){
+        if (dbs.length === 0) return;
+
+        let diastolicReadings = dbs.map(db => db.averageDiastolic);
+        let diastolicWeeklyAvg = diastolicReadings.reduce((a, b) => a + b) / diastolicReadings.length;
+        diastolicWeeklyAvg = Math.round(diastolicWeeklyAvg);
+        let diastolicWeeklyAvgAlert = getDiastolicCategory(diastolicWeeklyAvg);
+
+        let systolicReadings = dbs.map(db => db.averageSystolic);
+        let systolicWeeklyAvg = systolicReadings.reduce((a, b) => a + b) / systolicReadings.length;
+        systolicWeeklyAvg = Math.round(systolicWeeklyAvg);
+        let systolicWeeklyAvgAlert = getSystolicCategory(systolicWeeklyAvg);
+
+        let bpMetrics : BloodPressureMetrics = {
+            averageSystolic : systolicWeeklyAvg, 
+            averageSystolicAlert: Category[systolicWeeklyAvgAlert],
+            averageSystolicAlertClass : systolicWeeklyAvgAlert == Category.Normal ? "mdhui-blood-pressure-metric-normal" : "mdhui-blood-pressure-metric-not-normal",
+            averageDiastolic : diastolicWeeklyAvg,
+            averageDiastolicAlert: Category[diastolicWeeklyAvgAlert],
+            averageDiastolicAlertClass : diastolicWeeklyAvgAlert == Category.Normal ? "mdhui-blood-pressure-metric-normal" : "mdhui-blood-pressure-metric-not-normal",
+            minDiastolicAlertClass : "",
+            maxSystolicAlertClass : ""
+        };
+
+        let diastolicLows = dbs.map(db => db.dumbbell.dataPoint ? db.dumbbell.dataPoint.dataSet1.values[0] : Infinity);
+        let diastolicLow = diastolicLows ? Math.min(...diastolicLows) : Infinity;
```

Comment:
Why "infinity" as a default here and not null or undefined?

Reply:
I preferred to stick with the number data type vs any when using undefined.
I ended up removing a lot of this.

Reply:
it isn't "any" though, it would just be ```number | undefined``` or ```number | null```

Reply:
Math.min and Math.max were not happy being supplied number | undefined. 
I ended up filtering out the non numeric values, which helped keep the array purely numeric.